src/uct/base/uct_iface.c

```diff
@@ -857,6 +857,54 @@ void uct_iface_get_local_address(uct_iface_local_addr_ns_t *addr_ns,
     }
 }
 
+const char *uct_iface_get_sysfs_path(const char *dev_path, const char *dev_name,
+                                     char *path_buffer)
+{
+    const char *detected_type = NULL;
+    char device_file_path[PATH_MAX];
+    char *sysfs_realpath;
+    struct stat st_buf;
+    char *sysfs_path;
+    int ret;
+
+    /* PF: realpath name is of form /sys/devices/.../0000:03:00.0/infiniband/mlx5_0 */
+    /* SF: realpath name is of form /sys/devices/.../0000:03:00.0/<UUID>/infiniband/mlx5_0 */
```

Comment:
minor: can also mention some net/eth device, as this func is not just for infiniband now